central/vulnerabilityrequest/utils/util.go

```diff
@@ -236,6 +239,40 @@ func V2GlobalScope(scope *storage.VulnerabilityRequest_Scope) bool {
 		imgScope.GetTag() == common.MatchAll
 }
 
+// IsUpdateNoOp returns true if the exception update does not change the original exception.
+func IsUpdateNoOp(req *storage.VulnerabilityRequest, update *common.UpdateRequest) bool {
+	if req == nil || update == nil {
+		return true
+	}
+	if update.FalsePositiveUpdate == nil && update.DeferralUpdate == nil {
+		return true
+	}
+
+	existingCVEs := req.GetCves().GetCves()
+	if deferralUpdate := update.DeferralUpdate; deferralUpdate != nil {
+		newCVEs := deferralUpdate.GetCVEs()
+		return equal(existingCVEs, newCVEs) &&
+			reflect.DeepEqual(req.GetDeferralReq().GetExpiry(), deferralUpdate.GetExpiry())
+	}
+
+	if falsePositiveUpdate := update.FalsePositiveUpdate; falsePositiveUpdate != nil {
+		newCVEs := falsePositiveUpdate.GetCVEs()
+		return equal(existingCVEs, newCVEs)
+	}
+	// If we cannot determine that the update is a deferral update and false-positive, then we cannot handle it and must be treated as noop.
```

Comment:
nit:  is it worth a debug log for this case?

Reply:
I will add a panic on dev builds because we would reach here if we change the struct that causes regression.